From e6f1e2c27734f6323b8207e19ea5ea26cfc4cb08 Mon Sep 17 00:00:00 2001
From: Nicolas Dufresne <nicolas.dufresne@collabora.com>
Date: Mon, 18 Dec 2017 11:34:13 -0500
Subject: [PATCH 5/5] meson: Enable building LV2

Note that I only support lilv 0.22+ for now, because it's simple and old
enough. If anyone needs to use the 0.16 support it will need to be
added.
---
 ext/lv2/meson.build | 19 +++++++++++++++++++
 ext/meson.build     |  2 +-
 2 files changed, 20 insertions(+), 1 deletion(-)
 create mode 100644 ext/lv2/meson.build

diff --git a/ext/lv2/meson.build b/ext/lv2/meson.build
new file mode 100644
index 000000000..bcb6442d8
--- /dev/null
+++ b/ext/lv2/meson.build
@@ -0,0 +1,19 @@
+lv2_sources = [
+  'gstlv2.c',
+  'gstlv2filter.c',
+  'gstlv2source.c',
+  'gstlv2utils.c',
+]
+
+lv2_dep = dependency('lilv-0', version : '>= 0.22', required : false)
+if lv2_dep.found()
+  gstrsvg = library('gstlv2',
+    lv2_sources,
+    c_args : gst_plugins_bad_args,
+    link_args : noseh_link_args,
+    include_directories : [configinc],
+    dependencies : [gstaudio_dep, gstbase_dep, lv2_dep, mathlib],
+    install : true,
+    install_dir : plugins_install_dir,
+  )
+endif
diff --git a/ext/meson.build b/ext/meson.build
index b14054f8a..0ba5608d3 100644
--- a/ext/meson.build
+++ b/ext/meson.build
@@ -24,7 +24,7 @@ subdir('kate')
 subdir('ladspa')
 subdir('libde265')
 subdir('libmms')
-#subdir('lv2')
+subdir('lv2')
 #subdir('modplug')
 subdir('mpeg2enc')
 subdir('mplex')
-- 
2.16.3

